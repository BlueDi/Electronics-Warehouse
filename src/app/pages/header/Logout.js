import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { withCookies } from 'react-cookie';
import { Label, Menu } from 'semantic-ui-react';
import { service } from '@utils';

class Logout extends Component {
  handleLogout = () => {
    service
      .post('/logout')
      .then(response => {
        response.data != 0
          ? this.props.history.push('/')
          : console.warn('Invalid logout');
      })
      .catch(error => {
        throw error;
      });
  };

  render() {
    var userCart = this.props.cookies.get('cart') || [];
    var userID = this.props.cookies.get('id') || undefined;
    var userName = this.props.cookies.get('name') || undefined;
    var cartLength = userCart.length;
    var emptyCart = cartLength <= 0;

    return (
      <Menu.Menu position="right">
        <Menu.Item as={Link} to={'/user/' + userID}>
          {userName}
          {!emptyCart && <Label circular>{userCart.length}</Label>}
        </Menu.Item>
        <Menu.Item link onClick={this.handleLogout}>
          Logout
        </Menu.Item>
      </Menu.Menu>
    );
  }
}

export default withRouter(withCookies(Logout));
