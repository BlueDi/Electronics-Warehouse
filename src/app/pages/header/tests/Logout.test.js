import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import Logout, { UserMenu } from '@pages/header/Logout';

describe('UserMenu', () => {
  it('should be defined', () => {
    expect(UserMenu).toBeDefined();
  });

  it('renders properly', () => {
    const component = renderer.create(
      <Router>
        <UserMenu userName={'Test'} userCartLength={1} userPath={'/test'} />
      </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Logout', () => {
  it('should be defined', () => {
    expect(Logout).toBeDefined();
  });

  it('renders properly', () => {
    const component = renderer.create(
      <Router>
        <Logout />
      </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
