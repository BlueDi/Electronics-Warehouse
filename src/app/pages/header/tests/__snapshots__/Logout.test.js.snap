// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> Renders properly 1`] = `
<div
  className="right menu"
>
  <a
    className="item"
    href="/requests_list"
    onClick={[Function]}
  >
    Requests
  </a>
  <a
    className="item"
    href=""
    onClick={[Function]}
  />
  <a
    className="link item"
    onClick={[Function]}
  >
    Logout
  </a>
</div>
`;
